xcc/commands.py

```diff
@@ -251,7 +253,22 @@ def get_job(
     if status:
         return job.status
     if result:
-        return job.get_result(integer_overflow_protection=False)
+        result = job.get_result(integer_overflow_protection=False)
+
+        def convert_to_list(value: Union[np.ndarray, List[np.ndarray]]) -> List:
+            """Converts the given job result value into a native Python list."""
+            if isinstance(value, np.ndarray):
+                return value.tolist()
+
+            return [array.tolist() for array in value]
+
+        # Convert all NumPy arrays into lists to take advantage of pprint.
+        list_result = {key: convert_to_list(value) for key, value in result.items()}
```

Comment:
Not necessarily an improvement, but is there a benefit to this instead of casting to and back from an array?
```suggestion
        # Convert all NumPy arrays into lists to take advantage of pprint.
        list_result = {key: np.array(value).tolist() for key, value in result.items()}
```
Wouldn't work for differently sized arrays, which could be the case for the metadata, I guess. 🤔

Reply:
This is a nice idea but, unfortunately, we can't assume each item in `result["output"]` has the same shape. 🙁

Reply:
Fair enough. I thought that might be the case.